Kavita.Common/Configuration.cs

```diff
@@ -140,4 +146,54 @@ private static int GetPort(string filePath)
     }
 
     #endregion
+
+    #region Ip Addresses
+
+    private static void SetIpAddresses(string filePath, string ipAddresses)
+    {
+        if (new OsInfo(Array.Empty<IOsVersionAdapter>()).IsDocker)
+        {
+            return;
+        }
+
+        try
+        {
+            var currentIpAddresses = GetIpAddresses(filePath);
+            var json = File.ReadAllText(filePath).Replace("\"IpAddresses\": \"" + currentIpAddresses, "\"IpAddresses\": \"" + ipAddresses);
```

Comment:
Changing the IP address with the UI will not work on existing installations because the string replace does nothing if the appsettings.json does not contain the setting. 
Same is true for the port if it was manually changed in the appsettings.json to 'Port:5000' (no space after : ).
I would change the config set methods to use json de-/serialization instead if this is okay for you?

Reply:
If you know how, I would love it. My config code is very hacky.